common/amcl-extensions/ecp_ZZZ.c

```diff
@@ -121,12 +126,54 @@ int32_t ecp_ZZZ_fromhash(ECP_ZZZ *point_out, const uint8_t *message, uint32_t me
     return -1;
 }
 
+static
+uint8_t get_random_byte(uint8_t *pool,
+                        size_t pool_size,
+                        uint8_t **pool_ptr,
+                        void (*get_random)(void *buf, size_t buflen))
+{
+    if (*pool_ptr >= (pool + pool_size)) {
+        get_random(pool, pool_size);
+        *pool_ptr = pool;
+    }
+
+    *pool_ptr += 1;
+    return *(*pool_ptr - 1);
+}
+
 void ecp_ZZZ_random_mod_order(BIG_XXX *big_out,
-                              csprng *rng)
+                              void (*get_random)(void *buf, size_t buflen))
 {
+    // 1) Get the order of the group
     BIG_XXX curve_order;
     BIG_XXX_rcopy(curve_order, CURVE_Order_ZZZ);
 
-    BIG_XXX_randomnum(*big_out, curve_order, rng);
+    // 2) Fill our pool with randomness (will be replenished if needed)
```

Comment:
What's the purpose of the pool?  Just an speed optimization versus doing `get_random(&r, 1)` in the main loop?

The pool management stuff seems out place in this function and the `get_random_byte` function out of place in this file.  Should the generic pool management stuff be in `ecdaa/randomness.h`?

Reply:
Yea, the pool is an optimization. 

I can put the pool management stuff in a separate `common/internal-utils` file. But, since it's for our internal use and not for the end user, I don't think it should go in `randomness.h`.